File: SparkPrograms2/src/main/java/com/siva/spark/HelloWorld.java
package com.siva.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

/**
 * Created by sivakumaran on 9/27/2016.
 */
public class HelloWorld {
    public static void main(String args[]){
        SparkConf conf = new SparkConf();
        conf.setMaster("local");
        conf.setAppName("Spark Test in Java");
        JavaSparkContext javaSparkContext  = new JavaSparkContext(conf);
        JavaRDD<String> lines = javaSparkContext.textFile("test.txt");
        System.out.println(lines.count());
        System.out.println(lines.collect());
        //System.out.println(lines.map());
    }
}

File: SparkPrograms2/src/main/scala/HelloWorld.scala
/**
  * Created by siva on 4/30/16.
  */
import org.apache.spark.{SparkConf, SparkContext}

object HelloWorld extends App{

  //def main(args:Array[String]): Unit ={
    println("Hello World")

    val sc = new SparkContext(new SparkConf().setMaster("local"))
    var l = sc.parallelize(List("hello Spark","Map Reduce"))
    println(l.map(x=>x.split(" ")))
    println(l.flatMap(x=>x.split(" ")))
    l.flatMap(_.split(" ")).foreach(println)


  //}

}
